src/apps/vpn/inspector/inspectorhandler.cpp

```diff
@@ -169,6 +169,63 @@ static QList<InspectorCommand> s_commands{
                        return obj;
                      }},
 
+    InspectorCommand{"query", "Query the tree", 1,
+                     [](InspectorHandler*, const QList<QByteArray>& arguments) {
+                       QJsonObject obj;
+                       obj["value"] =
+                           !!InspectorUtils::queryObject(arguments[1]);
+                       return obj;
+                     }},
+
+    InspectorCommand{
+        "property_by_query", "Retrieve a property value from an object", 2,
+        [](InspectorHandler*, const QList<QByteArray>& arguments) {
+          QJsonObject obj;
+
+          QObject* item = InspectorUtils::queryObject(arguments[1]);
+          if (!item) {
+            obj["error"] = "Object not found";
+            return obj;
+          }
+
+          QVariant property = item->property(arguments[2]);
+          if (!property.isValid()) {
+            obj["error"] = "Property is invalid";
+            return obj;
+          }
+
+          obj["value"] = property.toString();
+          return obj;
+        }},
+
+    InspectorCommand{
+        "set_property_by_query", "Set a property value to an object", 4,
+        [](InspectorHandler*, const QList<QByteArray>& arguments) {
+          QJsonObject obj;
+
+          QVariant value;
+          if (arguments[3] == "i") {
+            value = arguments[4].toInt();
+          } else if (arguments[3] == "s") {
+            value = arguments[4];
```

Comment:
Quick Nit Suggestion: given this is currently on the table to get update -  maybe the settings-property setter can be reused instead of requiring the caller to know the type. 

```
          QMetaProperty property = metaObject->property(propertyId);
          Q_ASSERT(property.isValid());

          QVariant value = QVariant::fromValue(arguments[2]);
          if (value.canConvert(property.type())) {
            property.write(settingsHolder, value);
            return obj;
          }
```